api_docs/trashbin.apib

```diff
@@ -0,0 +1,344 @@
+* 404: Item does not exist in the trash bin
+
++ Response 201 (application/json)
+
+        {
+            "kind": "dds-folder",
+            "id": "d5ae02a4-b9e6-473d-87c4-66f4c881ae7a",
+            "parent": {
+                "kind": "dds-folder",
+                "id": "552be8c5-209d-4e3b-afaf-cb66686ffbyy"
+            },
+            "name": "Raw Data",
+            "project": {
+                "id": "ca29f7df-33ca-46dd-a015-92c46fdb6fd1"
+            },
+            "ancestors": [
+                {
+                    "kind": "dds-project",
+                    "id": "ca29f7df-33ca-46dd-a015-92c46fdb6fd1",
+                    "name": "Knockout Mouse Project (KOMP)"
+                },
+                {
+                    "kind": "dds-folder",
+                    "id": "552be8c5-209d-4e3b-afaf-cb66686ffbyy",
+                    "name": "Sequencing Core"
+                }
+            ],
+            "is_deleted": false,
+            "audit": { }
+        }
+
+## Purge Trashbin Item [/trashbin/{object_id}/{object_kind}/purge]
++ Parameters
+    + object_kind (string, required, `dds-file`) ... The kind of object.
+    + object_id (string, required, `777be35a-98e0-4c2e-9a17-7bc009f9b111`) ... The unique id of the object.
+
+### Purge Trashbin Item instance [PUT]
+>**Permission:** create_file or system_admin
+- - - - -
+**Response messages:**
+* 201: Valid API Token in 'Authorization' Header
+* 400: Parent object does not exist, has been purged, or is in the trash bin
+* 401: Missing, Expired, or Invalid API Token in 'Authorization' Header
+* 403: Forbidden
+* 404: Item does not exist in the trash bin
+
++ Response 201
+
+## Project Trashbin Items [/trashbin/project/{id}/children{?name_contains,recurse}]
++ Parameters
+    + id (string, required, `777be35a-98e0-4c2e-9a17-7bc009f9b111`) ... The unique id of the project.
+    + name_contains (string, optional)
+    + recurse (boolean, optional)
+
+### Project Trashbin Items instance [GET]
+>**Permission:** view_project or system_admin
+- - - - -
+
+>**Rules**
+- - - - -
++ Lists all children of the project unless the `name_contains` parameter is specified; when specified, only childen whose name contains the `name_contains` parameter are returned.
++ Lists only immediate children of the project unless the `recurse` parameter is specified, when specified, invokes a recursive search of the project hierarchy.
+
+**Response messages:**
+* 201: Valid API Token in 'Authorization' Header
+* 400: Parent object does not exist, has been purged, or is in the trash bin
+* 401: Missing, Expired, or Invalid API Token in 'Authorization' Header
+* 403: Forbidden
+* 404: Item does not exist in the trash bin
+
++ Response 200 (application/json)
+
+        {
+            "results": [
+                {
+                    "kind": "dds-folder",
+                    "id": "d5ae02a4-b9e6-473d-87c4-66f4c881ae7a",
+                    "parent": {
+                        "kind": "dds-folder",
```

Comment:
I'd say that the default for this endpoint is `recurse: false`. SO, these examples should have the project as the `parent`.